packages/wells/README.md

```diff
@@ -53,6 +53,26 @@ client.loginWithApiKey({
 });
 ```
 
+### set up client with Token
+
+```ts
+import { createWellsClient, Cluster } from '@cognite/sdk-wells';
+
+// Cluster.API (default), Cluster.BP, Cluster.GREENFIELD
+let client = createWellsClient('app id', Cluster.API);
+
+client.loginWithToken({
+  project: process.env.COGNITE_WELLS_PROJECT,
+  accessToken: '*YOUR-TOKEN-HERE*',
+});
+
+// whenever you token is updated
+client.loginWithToken({
+  project: process.env.COGNITE_WELLS_PROJECT,
+  accessToken: '*UPDATE-WITH-NEW-TOKEN-HERE*',
+});
+```
```

Comment:
As opposed to an api-key, a token will expire and you would have to refresh and provide a new token on a regular basis. The core cdf js sdk seems to update this token automatically in the background using a handler `set401ResponseHandler` for **401** response, which then tries to re-authenticate using the `/login/redirect` flow in [CDF Auth](https://docs.cognite.com/api/v1/#section/Authentication) to obtain a bearer access token when a 401 has happend. Authenticate method is created by [createAuthenticateFunction](https://github.com/cognitedata/cognite-sdk-js/blob/de3557f0b4ffb62d2ccdf3145d38108554a82763/packages/core/src/login.ts#L304) function, where bearer token is going to be set.

In the well service we don't have a way to provide new tokens to the client, thus the token has to be provided by the user themselves. Both login and updating the token is done calling `loginWithToken({...})`. This might be an anti-pattern, and mabye we should rather have a retry-mechanism in the http calls that tries to re-authenticate whenever a 401 response is thrown by the well service API.

@jucc @drsphelps

Reply:
@rek @deep-cognite  Any thoughts here?

Reply:
@Sollimann  what if you allowed us to provide any function to createWellsClient method that will be executed when you get 401 error in the sdk. Then we can handle relogging  from our end. But not sure this is the best solution :)

Reply:
@rumeshwick I like that idea. Will try to add that :) I'll probably allow this function to be inserted into the `client.loginWithToken method` instead of with the createWellsClient, since this method is not needed when authenticating with an api-key.